### ISSUE TITLE
install.sh doesn't stop if wget is not available

### ISSUE THREAD

**Issue by InfernoZeus** [slap-editor/slap#177]
Inside of an Ubuntu 14.04.2 Docker container, both curl and wget are unavailable. I manually installed curl to download the install script, but there's no indication that wget is needed as well. It then fails when trying to add the NodeSource repo to dpkg:
```
ben@ap:~$ curl -sL https://raw.githubusercontent.com/slap-editor/slap/master/install.sh | sh
# Installing slap...
sh: 21: wget: not found
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following extra packages will be installed:
  git-man libc-ares2 libcurl3-gnutls libedit2 liberror-perl libv8-3.14.5
  openssh-client patch rsync xauth
Suggested packages:
  gettext-base git-daemon-run git-daemon-sysvinit git-doc git-el git-email
  git-gui gitk gitweb git-arch git-bzr git-cvs git-mediawiki git-svn
  ssh-askpass libpam-ssh keychain monkeysphere ed diffutils-doc openssh-server
Recommended packages:
  ssh-client
The following NEW packages will be installed:
  git git-man libc-ares2 libcurl3-gnutls libedit2 liberror-perl libv8-3.14.5
  nodejs openssh-client patch rsync xauth
0 upgraded, 12 newly installed, 0 to remove and 5 not upgraded.
Need to get 6470 kB of archives.
After this operation, 34.4 MB of additional disk space will be used.
Get:1 http://ubuntu-archive.mirror.nucleus.be/ trusty-updates/main libcurl3-gnutls amd64 7.35.0-1ubuntu2.5 [165 kB]
Get:2 http://ubuntu-archive.mirror.nucleus.be/ trusty/main libedit2 amd64 3.1-20130712-2 [86.7 kB]
Get:3 http://ubuntu-archive.mirror.nucleus.be/ trusty/main libc-ares2 amd64 1.10.0-2 [38.5 kB]
Get:4 http://ubuntu-archive.mirror.nucleus.be/ trusty-updates/main openssh-client amd64 1:6.6p1-2ubuntu2 [565 kB]
Get:5 http://ubuntu-archive.mirror.nucleus.be/ trusty-updates/main rsync amd64 3.1.0-2ubuntu0.1 [283 kB]
Get:6 http://ubuntu-archive.mirror.nucleus.be/ trusty/main xauth amd64 1:1.0.7-1ubuntu1 [25.8 kB]
.....
```
install.sh should probably error and exit instead of continuing if it fails to download the repo.

**Comment by dbkaplun** [comment]
Good point! The install script simply installs `node` and `npm`, so you don't need the script if you already have them both -- just run `npm install -g slap`. (You might need `git` as well.)

About the error, not sure why this is happening... [`set -e`](https://github.com/slap-editor/slap/blob/master/install.sh#L3) is supposed to cause the command to exit if any subcommand exits with nonzero status. What shell are you using?

**Comment by InfernoZeus** [comment]
I think it's `bash`. I'll have a check in a couple of hours when I've got some free time.

**Comment by InfernoZeus** [comment]
With a bit of trial and error (and a bit of Googling), it seems as if you need to use `set -o pipefail` otherwise the return value of a pipe is the return value of the last command in the chain. In our case, `sudo bash -` returns a success as nothing is piped into it. Unfortunately `pipefail` is a bashism.

Unfortunately, there doesn't seem to be an amazing solution for plain old `sh`. The comp.unix.shell FAQ suggests this function for POSIX shells:
```
       run() {
         j=1
         while eval "\${pipestatus_$j+:} false"; do
           unset pipestatus_$j
           j=$(($j+1))
         done
         j=1 com= k=1 l=
         for a; do
           if [ "x$a" = 'x|' ]; then
             com="$com { $l "'3>&-
                         echo "pipestatus_'$j'=$?" >&3
                       } 4>&- |'
             j=$(($j+1)) l=
           else
             l="$l \"\$$k\""
           fi
           k=$(($k+1))
         done
         com="$com $l"' 3>&- >&4 4>&-
                    echo "pipestatus_'$j'=$?"'
         exec 4>&1
         eval "$(exec 3>&1; eval "$com")"
         exec 4>&-
         j=1
         while eval "\${pipestatus_$j+:} false"; do
           eval "[ \$pipestatus_$j -eq 0 ]" || return 1
           j=$(($j+1))
         done
         return 0
       }
       
       use it as:
       
       run cmd1 \| cmd2 \| cmd3
       exit codes are in $pipestatus_1, $pipestatus_2, $pipestatus_3
```

[A StackOverflow answer](http://stackoverflow.com/a/1550982) suggests temporary files might be there easiest way. I can submit a pull request for something along those lines if you want?

**Comment by InfernoZeus** [comment]
Alternatively, just use `cURL` like the `yum` section of `install.sh` does. That's probably the best solution actually. I'll submit a pull request for that.

**Comment by dbkaplun** [comment]
Closed as of #178. Great work @InfernoZeus!
